homeassistant/components/sensor/emoncms.py

```diff
@@ -0,0 +1,246 @@
+                        name = include_feeds_names[listindex]
+
+                sensors.append(EmonCmsSensor(hass, data, name,
+                                             value_template,
+                                             unit_of_measurement,
+                                             str(sensorid),
+                                             elem))
+    add_devices(sensors)
+
+
+# pylint: disable=too-many-instance-attributes
+class EmonCmsSensor(Entity):
+    """Implementation of an EmonCmsSensor sensor."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, hass, data, name, value_template,
+                 unit_of_measurement, sensorid, elem):
+        """Initialize the sensor."""
+        if name is None:
+            self._name = "emoncms" + sensorid + "_feedid_" + elem["id"]
+        else:
+            self._name = name
+        self._identifier = get_id(sensorid, elem["tag"],
+                                  elem["name"], elem["id"],
+                                  elem["userid"])
+        self._hass = hass
+        self._data = data
+        self._value_template = value_template
+        self._unit_of_measurement = unit_of_measurement
+        self._sensorid = sensorid
+        self._feedid = elem["id"]
+        self._userid = elem["userid"]
+        self._tag = elem["tag"]
+        self._feedname = elem["name"]
+        self._size = elem["size"]
+        self._lastupdatetime = elem["time"]
+        self._value = elem["value"]
+        if self._value_template is not None:
+            self._state = template.render_with_possible_json_value(
+                self._hass, self._value_template, self._value,
+                STATE_UNKNOWN)
+        else:
+            self._state = round(float(self._value), DECIMALS)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the atrributes of the sensor."""
+        return {
+            ATTR_FEEDID: self._feedid,
+            ATTR_TAG: self._tag,
+            ATTR_FEEDNAME: self._feedname,
+            ATTR_SIZE: self._size,
+            ATTR_USERID: self._userid,
+            ATTR_LASTUPDATETIME: self._lastupdatetime,
+            ATTR_LASTUPDATETIMESTR: template.timestamp_local(
+                float(self._lastupdatetime)),
+        }
+
+    def update(self):
+        """Get the latest data and updates the state."""
+        if self._data is None:
+            return
+
+        found = False
+        if self._data.data is not None:
+            for elem in self._data.data:
+                elemid = get_id(self._sensorid, elem["tag"], elem["name"],
```

Comment:
It would be nice to build a dict in your update method below with get_id as the key. Then you only do it once for multiple sensors, but that's up to you!